### ISSUE TITLE
"Use primary constructor" does not produce a diagnostic for an empty public parameterless constructor

### ISSUE THREAD

**Issue by JakeYallop** [dotnet/roslyn#73695]
**Version Used**:  `Compiler version: 4.10.0-3.24230.3` (abc35d6d)

**Steps to Reproduce**:

```csharp
public class MyClass
{
    public MyClass()
    {
    }
}
```

**Diagnostic Id**: IDE0290

**Expected Behavior**:
IDE0290 "Use primary constructor" diagnostic is raised.

**Actual Behavior**:
No diagnostic is raised.

**Other notes**
Just adding a parameter to the constructor:
```csharp
public class MyClass
{
    public MyClass(string prop)
    {
    }
}
```
causes the diagnostic be raised.

**Comment by CyrusNajmabadi** [comment]
Agreed.

**Comment by DoctorKrolic** [comment]
> An empty primary constructor doesn't really help at all, since there are no parameters to capture or assign to fields.

If such constructor has attributes, it might be benifitial to make it a primary one. A common example pattern is an osbolete MEF import constructors in roslyn

**Comment by CyrusNajmabadi** [comment]
@DoctorKrolic i like that idea.  let me go do that.
